src/Validator/Rules/ValuesOfCorrectType.php

```diff
@@ -33,8 +35,15 @@
  */
 class ValuesOfCorrectType extends AbstractValidationRule
 {
-    static function badValueMessage($typeName, $valueName, $message = null)
+    private static $fieldName;
+
+    static function badValueMessage($typeName, $valueName, $message = null, $context = null)
     {
+        $fieldName = self::$fieldName;
+        if ($context AND $arg = $context->getArgument() AND $arg instanceof FieldArgument) {
+            $argName = $arg->name;
+            return "Field \"{$fieldName}\" argument \"{$argName}\" requires type {$typeName}, found {$valueName}" . ($message ? "; {$message}" : '.');
+        }
```

Comment:
This message should be in a separate function (like `badArgumentValue`). Also, those functions shouldn't contain any logic related to errors - only formatting logic. The logic of triggering an error should be defined in the visitor.

Reply:
Resolved in 9dc41f0